src/Section.cpp

```diff
@@ -224,18 +222,18 @@ std::vector<Section> Section::findDownstream(const std::function<bool(Section)>
 }
 
 
-std::vector<Section> Section::findUpstream(const std::function<bool(Section)> &filter) const{
+std::vector<Section> Section::findAmongParents(const std::function<bool(Section)> &filter) const {
     std::vector<Section> results;
     Section p = parent();
-
-    if (p != none) {
-        results = p.findSections(filter,1);
-        if (results.size() > 0) {
-            return results;
-        }
-        return p.findUpstream(filter);
+    if (p == none) {
+        return results;
+    }
+    if (filter(p)) {
+        results.push_back(p);
+        return results;
```

Comment:
So we always return exactly one? Or am I am missing something? You wanna keep the result type as a vector to stay compatible with the other search functions?

Reply:
Yes, in this Case we return at maximum one section. And yes, I wanted to keep it similar to the other functions. Happy to change, if you want.